### PR TITLE
Fix skipping of LuminosityBlock with firstLuminosityBlockForEachRun

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

CMSSW_11_1_0_pre6 showed that the tests for Run dependent MC had failed. This adds tests which exhibit the problem and provides changes which allow those tests to pass.

#### PR validation:

The code compiles and the new (and extended) tests all pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: IOPool/Input/src/RootFile.cc [hunk 1]
```diff
@@ -65,6 +65,7 @@
 
 #include <algorithm>
 #include <list>
+#include <iostream>
```
Comment (makortel) [inline IOPool/Input/src/RootFile.cc:68]:
```suggestion
```
This `#include` seems not to be needed.

File: IOPool/Input/test/firstLuminosityBlockForEachRun_skipLumis2_Test_cfg.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# The following comments couldn't be translated into the new config version:
+
+# Configuration file for PoolInputTest
+
+import FWCore.ParameterSet.Config as cms
+import sys
+
+import sys
```
Comment (makortel) [inline IOPool/Input/test/firstLuminosityBlockForEachRun_skipLumis2_Test_cfg.py:8]:
```suggestion
```
one `import` should be sufficient :)

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**